driver/src/orderbook/streamed/state.rs

```diff
@@ -198,6 +234,8 @@ impl State {
         sell_balance_fn: impl FnOnce(&mut Balance) -> Result<()>,
         buy_balance_fn: impl FnOnce(&mut Balance) -> Result<()>,
     ) -> Result<()> {
+        self.solution_partially_received = true;
```

Comment:
What is the meaning and purpose of a partially received solution? I see it is set true here and then false again later. Is this flag only accounting for the time between here and a few lines down the road? If this is the case, does this flag act as a sort-of lock for some sort of threaded environment?

Reply:
The intention is to notice if someone tries to get the orderbook from us while we have received only some trades of a solution but not all of them. In that case the orderbook wouldn't be correct.